cluster-autoscaler/cloudprovider/clusterapi/clusterapi_scalableresource.go

```diff
@@ -60,3 +67,23 @@ type scalableResource interface {
 	InstanceGPUCapacity() (resource.Quantity, error)
 	InstanceMaxPodsCapacity() (resource.Quantity, error)
 }
+
+func unmarkMachineForDeletion(controller *machineController, machine *Machine) error {
+	u, err := controller.dynamicclient.Resource(*controller.machineResource).Namespace(machine.Namespace).Get(context.TODO(), machine.Name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+	if u == nil {
+		return fmt.Errorf("unknown machine %s", machine.Name)
+	}
```

Comment:
Is this needed? Shouldn't err be a 404 if the machine wasn't found?

Reply:
good question, this is copied from the MarkMachineForDeletion function and it has the same logic.

Reply:
thinking on this a little more, i think you are correct that the error should be of some not found type if the machine was unknown.

that said, we would still want to check nil here but maybe we should have a different error message. wdyt?

Reply:
As far as I understand, if the err is nil, that means the request was successful and the response was unmarshalled into `u`, so I don't think we should need to check if u is nil or not, I don't mind leaving it for now, but I'd also suggest seeing if the tests fail if you remove it, I suspect not

From my understanding of the client at least, both u and err being nil would be a coding error and the sort of thing I would panic on normally, it should never happen 🤔

Reply:
that makes perfect sense to me. i kinda figured the same, but given that the other function carried it i figured i should just keep it for now.

could i just make a follow up pr to remove that check from both functions?

Reply:
Follow up sgtm